Modules/StudyProgramme/classes/types/class.ilStudyProgrammeTypeCustomIconsFormGUI.php

```diff
@@ -82,8 +83,9 @@ protected function initForm() {
 		$item = new ilImageFileInputGUI($this->lng->txt('icon'), 'icon');
 		$item->setSuffixes(array( 'svg' ));
 		$item->setInfo($this->lng->txt('prg_type_custom_icon_info'));
-		if (is_file($this->type->getIconPath(true))) {
-			$item->setImage($this->type->getIconPath(true));
+		if ($this->g_webdir->has($this->type->getIconPath(true))) {
+			// TODO: that´s horrible, try to avoid ilUtil in future
+			$item->setImage(ilUtil::getWebspaceDir().'/'.$this->type->getIconPath(true));
```

Comment:
This might be a feature request for the filesystem-abstraction, right?

Reply:
In my opinion, if every ILIAS class use the filesystem-abstraction everything is alright. Cause setImage() in this case don´t use file-abstraction we have to give it the full path to the icon.